[Bug] TypeScriptビルドエラーの修正

## 問題の概要
Vercelでのデプロイ時にTypeScriptビルドエラーが発生しています。

## エラー内容

### 1. database.ts - updatedAtプロパティのアクセス方法
```
src/data/database.ts(123,50): error TS4111: Property 'updatedAt' comes from an index signature, so it must be accessed with ['updatedAt'].
src/data/database.ts(135,50): error TS4111
src/data/database.ts(144,50): error TS4111
src/data/database.ts(152,50): error TS4111
```

### 2. api-key-test.ts - 型推論とprivateプロパティアクセス
```
src/services/api-key-test.ts(38,21): error TS2352: Conversion of type 'WeatherService' to type '{ apiKey?: string; }' may be a mistake
src/services/api-key-test.ts(157-182): error TS2345: Argument of type 'string' is not assignable to parameter of type 'never'
```

### 3. geolocation.service.test.ts - モック型の不一致
```
src/services/geolocation.service.test.ts(38,25): error TS2345: Type missing properties: timestamp, toJSON
src/services/geolocation.service.test.ts(55,23): error TS2345: Type missing properties: PERMISSION_DENIED, POSITION_UNAVAILABLE, TIMEOUT
```

### 4. notification-scheduler.service.ts - Record型の不一致
```
src/services/notification-scheduler.service.ts(445,29): error TS2345: Type '[]' is not assignable to parameter of type 'Record<string, unknown>'
```

### 5. weather.service.ts - leisureプロパティの不在
```
src/services/weather.service.ts(431,15): error TS2339: Property 'leisure' does not exist
```

### 6. test-utils.tsx - import構文
```
src/test-utils.tsx(3,18): error TS1484: 'RenderOptions' must be imported using a type-only import when 'verbatimModuleSyntax' is enabled
```

## 影響範囲
- Vercelでのビルドが失敗
- GitHub Actionsのビルドステップが失敗
- 型チェック (`npm run typecheck`) は通過
- ESLint (`npm run lint`) は通過

## 修正方針

### 優先度高
1. **database.ts**: インデックスシグネチャプロパティへのアクセスを `['updatedAt']` 形式に修正
2. **test-utils.tsx**: type-only importに修正
3. **api-key-test.ts**: 型アサーションの修正、recommendations配列の型定義を修正

### 優先度中
4. **geolocation.service.test.ts**: モック型を正しく定義
5. **notification-scheduler.service.ts**: details型をRecord型に適合させる

### 優先度低
6. **weather.service.ts**: leisureプロパティの型定義を追加

## 期待される効果
- Vercelでのビルド成功
- GitHub Actionsのビルドステップ成功
- TypeScript厳格モードでのエラー解消

## 関連情報
- PR #47でconsole.log削除作業を実施したが、これらのエラーは既存の問題
- mainブランチにも存在するビルドエラー

--- a/src/data/database.ts
+++ b/src/data/database.ts
@@ -120,7 +120,7 @@ export class HobbyWeatherDatabase extends Dexie {
     });
 
     this.hobbies.hook('updating', (modifications) => {
-      (modifications as Record<string, unknown>).updatedAt = new Date();
+      (modifications as Record<string, unknown>)['updatedAt'] = new Date();
     });
 
     this.locations.hook('creating', (_, obj) => {
@@ -132,7 +132,7 @@ export class HobbyWeatherDatabase extends Dexie {
     });
 
     this.settings.hook('updating', (modifications) => {
-      (modifications as Record<string, unknown>).updatedAt = new Date();
+      (modifications as Record<string, unknown>)['updatedAt'] = new Date();
     });
 
     this.notificationConfigs.hook('creating', (_, obj) => {
@@ -141,7 +141,7 @@ export class HobbyWeatherDatabase extends Dexie {
     });
 
     this.notificationConfigs.hook('updating', (modifications) => {
-      (modifications as Record<string, unknown>).updatedAt = new Date();
+      (modifications as Record<string, unknown>)['updatedAt'] = new Date();
     });
 
     this.notificationSettings.hook('creating', (_, obj) => {
@@ -149,7 +149,7 @@ export class HobbyWeatherDatabase extends Dexie {
     });
 
     this.notificationSettings.hook('updating', (modifications) => {
-      (modifications as Record<string, unknown>).updatedAt = new Date();
+      (modifications as Record<string, unknown>)['updatedAt'] = new Date();
     });
   }
 

--- a/src/services/api-key-test.ts
+++ b/src/services/api-key-test.ts
@@ -35,7 +35,7 @@ export async function checkWeatherServiceApiKey() {
     const { weatherService } = await import('./weather.service');
     
     // プライベートプロパティにアクセスできるように型キャスト
-    const service = weatherService as { apiKey?: string };
+    const service = weatherService as unknown as { apiKey?: string };
     
     console.log('WeatherService apiKey:', service.apiKey);
     console.log('WeatherService apiKey type:', typeof service.apiKey);
@@ -149,7 +149,7 @@ export async function runDiagnostics() {
     environment: envCheck,
     weatherService: serviceCheck,
     apiConnection: connectionCheck,
-    recommendations: []
+    recommendations: [] as string[]
   };
   
   // 推奨事項の生成
@@ -183,7 +183,7 @@ export async function runDiagnostics() {
   }
   
   console.log('推奨事項:');
-  results.recommendations.forEach((rec: string) => console.log(rec));
+  results.recommendations.forEach((rec) => console.log(rec));
   
   return results;
 }

--- a/src/services/geolocation.service.test.ts
+++ b/src/services/geolocation.service.test.ts
@@ -26,12 +26,19 @@ describe('GeolocationService', () => {
 
   describe('getCurrentPosition', () => {
     it('should get current position successfully', async () => {
-      const mockPosition = {
+      const mockPosition: GeolocationPosition = {
         coords: {
           latitude: 35.6762,
           longitude: 139.6503,
-          accuracy: 10
-        }
+          accuracy: 10,
+          altitude: null,
+          altitudeAccuracy: null,
+          heading: null,
+          speed: null,
+          toJSON: () => ({})
+        },
+        timestamp: Date.now(),
+        toJSON: () => ({})
       };
 
       mockGeolocation.getCurrentPosition.mockImplementation((successCallback: (position: GeolocationPosition) => void) => {
@@ -46,9 +53,12 @@ describe('GeolocationService', () => {
     });
 
     it('should handle permission denied error', async () => {
-      const mockError = {
+      const mockError: GeolocationPositionError = {
         code: 1,
-        message: 'User denied the request for Geolocation.'
+        message: 'User denied the request for Geolocation.',
+        PERMISSION_DENIED: 1,
+        POSITION_UNAVAILABLE: 2,
+        TIMEOUT: 3
       };
 
       mockGeolocation.getCurrentPosition.mockImplementation((_: (position: GeolocationPosition) => void, errorCallback: (error: GeolocationPositionError) => void) => {
@@ -62,9 +72,12 @@ describe('GeolocationService', () => {
     });
 
     it('should handle position unavailable error', async () => {
-      const mockError = {
+      const mockError: GeolocationPositionError = {
         code: 2,
-        message: 'Position unavailable.'
+        message: 'Position unavailable.',
+        PERMISSION_DENIED: 1,
+        POSITION_UNAVAILABLE: 2,
+        TIMEOUT: 3
       };
 
       mockGeolocation.getCurrentPosition.mockImplementation((_: (position: GeolocationPosition) => void, errorCallback: (error: GeolocationPositionError) => void) => {
@@ -78,9 +91,12 @@ describe('GeolocationService', () => {
     });
 
     it('should handle timeout error', async () => {
-      const mockError = {
+      const mockError: GeolocationPositionError = {
         code: 3,
-        message: 'Timeout expired.'
+        message: 'Timeout expired.',
+        PERMISSION_DENIED: 1,
+        POSITION_UNAVAILABLE: 2,
+        TIMEOUT: 3
       };
 
       mockGeolocation.getCurrentPosition.mockImplementation((_: (position: GeolocationPosition) => void, errorCallback: (error: GeolocationPositionError) => void) => {

--- a/src/services/notification-scheduler.service.ts
+++ b/src/services/notification-scheduler.service.ts
@@ -442,7 +442,9 @@ export class NotificationSchedulerService {
                             activeAlert.severity,
                             activeAlert.alertType,
                             activeAlert.message,
-                            activeAlert.details
+                            activeAlert.details.length > 0
+                                ? { items: activeAlert.details } as Record<string, unknown>
+                                : undefined
                         );
                 } else {
                     // アクティブなアラートがない場合はダミーまたはスキップ

--- a/src/services/weather.service.ts
+++ b/src/services/weather.service.ts
@@ -412,6 +412,7 @@ export class WeatherService {
         amenity?: string;
         shop?: string;
         tourism?: string;
+        leisure?: string;
       };
     }
 

--- a/src/test-utils.tsx
+++ b/src/test-utils.tsx
@@ -1,6 +1,6 @@
 /* eslint-disable react-refresh/only-export-components */
 import React from 'react';
-import { render, RenderOptions } from '@testing-library/react';
+import { render, type RenderOptions } from '@testing-library/react';
 import { ThemeProvider } from './contexts/ThemeContext';
 
 const AllTheProviders = ({ children }: { children: React.ReactNode }) => {

